_source/_posts/2025-07-18-react-pwa.md

```diff
@@ -0,0 +1,832 @@
+}
+
+export default Home;
+```
+
+This is a minimal home component that represents our home page. 
+
+## Authenticate using OAuth 2.0 and OpenID Connect (OIDC)
+
+Next, we want to add the ability to sign in and out with our Okta without social login as a starting point. We'll add the social login connection later.
+
+To do that, we'll create the `SignIn` component and a generic `Layout` component to control user access based on their authentication. Navigate to your `src` folder, then create a `components` folder to hold child components. 
+
+In the newly created ' components ' folder, create the `Layout.tsx`, `Layout.css`, and `SignIn.tsx` files.
+
+Open the `Layout.tsx` file and add the following code:
+
+```tsx
+import './Layout.css';
+import { useOktaAuth } from "@okta/okta-react";
+import SignIn from "./SignIn.tsx";
+import { Link } from "react-router-dom";
+import logo from '../assets/react.svg';
+
+const Layout = ({children}) => {
+ const { authState, oktaAuth} = useOktaAuth();
+ const signout = async () => await oktaAuth.signOut();
+
+ return authState?.isAuthenticated ? (<>
+  <div className="navbar">
+   <Link to="/"><img src={logo} className="logo" /></Link>
+   <div className="right">
+    <Link to="/profile">Profile</Link>
+    <button onClick={signout} className="no-outline">Sign Out</button>
+   </div>
+  </div>
+  <div className="layout">
+   {...children}
+  </div>
+ </>) : <SignIn/>;
+}
+export default Layout;
+```
+This component imports the `useOktaAuth` from the `@okta/okta-react` package. This React hook helps us the user's authenticated state and gives them access to the child components of the `Layout` component. The hook also lets us sign in or out our users.
+
+At the top of the file, we import `Layout.css`. Open `Layout.css` so fill in the CSS we need:
+
+```css
+.layout {
+ max-width: 1280px;
+ margin: 0 auto;
+ padding: 2rem;
+ text-align: center;
+}
+
+.layout.sign-in {
+ margin-top: 35vh;
+}
+
+.logo {
+ height: 32px;
+ will-change: filter;
+ transition: filter 300ms;
+}
+
+.navbar {
+ display: flex;
+ justify-content: space-between;
+}
+```
+
+These minor stylings help the `Layout.tsx` navbar look proper. Let's not forget the `SignIn` component used in the `Layout` component. 
+
+Paste the following code into `SignIn.tsx`:
+
+```tsx
+import { useOktaAuth } from "@okta/okta-react";
+import logo from '../assets/react.svg';
+
+const SignIn = () => {
+ const { oktaAuth} = useOktaAuth();
+ const signin = async () => await oktaAuth.signInWithRedirect();
+
+ return (
+  <div className="sign-in layout">
+  <h2> <img src={logo} className="logo" alt="Logo"/> Lister</h2>
+  <button className="outlined" onClick={signin}>Sign In</button>
+  </div>
+ );
+}
+
+export default SignIn;
+```
+
+Here, we use the same `useOktaAuth` hook to sign in our user. Lastly, we update `src/App.tsx` to use our new `Layout` component. We wrap the Layout component around the routes that require authentication. Your code should look like this:
```

Comment:
Replace the existing line with this one -> The code now looks like this:

(Please make similar swaps for any remaining “should” instances)